core/server/themes/loader.js

```diff
@@ -0,0 +1,39 @@
+var debug = require('debug')('ghost:themes:loader'),
+    config = require('../config'),
+    events = require('../events'),
+    read = require('./read'),
+    settingsApi = require('../api/settings'),
+    updateConfigAndCache,
+    loadThemes,
+    initThemes;
+
+updateConfigAndCache = function updateConfigAndCache(themes) {
+    debug('loading themes', themes);
+    config.set('paths:availableThemes', themes);
+    settingsApi.updateSettingsCache();
+};
+
+loadThemes = function loadThemes() {
+    return read
+        .all(config.getContentPath('themes'))
+        .then(updateConfigAndCache);
+};
+
+initThemes = function initThemes() {
+    debug('init themes', settingsApi.cache.get('activeTheme'));
+
+    // Register a listener for server-start to load all themes
+    events.on('server:start', function readAllThemesOnServerStart() {
```

Comment:
I think this could go either way.

This is information we need to load at some point - loading it after the server starts ensures it's present for when it is requested.

Leaving it as-is means there is an edge case where the theme API is requested in the gap between server start & themes being read and so the API would return just the active theme. I think if that happened a user would just refresh, and then get the right results.

If we move it onto being read on first request, that means that first request could be potentially very slow. Then we'd potentially be back to it taking too long for the server to respond and getting errors on Ghost(Pro) again?